addons/website/models/ir_ui_menu.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, tools
+from odoo.http import request
+
+
+class IrUiMenu(models.Model):
+    _inherit = 'ir.ui.menu'
+
+    @api.model
+    @tools.ormcache_context('self._uid', keys=('lang', 'force_action',))
+    def load_menus_root(self):
+        root_menus = super().load_menus_root()
+        if self._context.get('force_action'):
```

Comment:
I always wonder: is not the "right" way to access using `.env.context` and `._context` is there for compatibility?

Reply:
Hm. I guess[ it is](https://github.com/odoo/odoo/blame/c30c1dac33da05a4064d5353215098b2a5b3c87d/odoo/models.py#L5112-L5115) even though it's still (more than) widely used.
I am not sure I knew about it or if I forgot, but I replaced it.

https://github.com/odoo/odoo/blob/c30c1dac33da05a4064d5353215098b2a5b3c87d/odoo/models.py#L5112-L5115